backend/src/crd/controller/scheduledworkflow/controller.go

```diff
@@ -457,65 +465,107 @@ func (c *Controller) submitNextWorkflowIfNeeded(ctx context.Context, swf *util.S
 	log.WithFields(log.Fields{
 		ScheduledWorkflow: swf.Name,
-		Workflow:          workflow.Get().Name,
+		Workflow:          workflowName,
 	}).Infof("Submitting workflow for ScheduledWorkflow (%v): workflow (%v) successfully submitted (scheduled at: %v)",
-		swf.Name, workflow.Get().Name, commonutil.FormatTimeForLogging(nextScheduledEpoch))
-	return workflow, nextScheduledEpoch, nil
+		swf.Name, workflowName, commonutil.FormatTimeForLogging(nextScheduledEpoch))
+	return submitted, nextScheduledEpoch, nil
 }
 
 func (c *Controller) submitNewWorkflowIfNotAlreadySubmitted(
 	ctx context.Context,
 	swf *util.ScheduledWorkflow, nextScheduledEpoch int64, nowEpoch int64) (
-	*commonutil.Workflow, error) {
+	bool, string, error) {
 
 	workflowName := swf.NextResourceName()
 
 	// Try to fetch this workflow
 	// If it already exists, it means that it was already created in a previous iteration
 	// of this controller but that the controller failed to save this data.
-	foundWorkflow, isNotFoundError, err := c.workflowClient.Get(swf.Namespace,
+	_, isNotFoundError, err := c.workflowClient.Get(swf.Namespace,
 		workflowName)
 	if err == nil {
 		// The workflow was already created by a previous iteration of this controller.
 		// Nothing to do except returning the information needed by the controller to update
 		// the ScheduledWorkflow status.
-		return foundWorkflow, nil
+		return true, workflowName, nil
 	}
 
 	if !isNotFoundError {
 		// There was an error while attempting to retrieve the workflow
-		return nil, err
+		return false, "", err
 	}
 
-	// If the workflow is not found, we need to create it.
-	newWorkflow, err := swf.NewWorkflow(nextScheduledEpoch, nowEpoch)
-	createdWorkflow, err := c.workflowClient.Create(ctx, swf.Namespace, newWorkflow)
+	if swf.Spec.PipelineID != nil {
+
+		log.WithFields(log.Fields{
+			ScheduledWorkflow: swf.Name,
+			Workflow:          workflowName,
+		}).Infof("PipelineID used when submitting run",
+			swf.Name, workflowName, commonutil.FormatTimeForLogging(nextScheduledEpoch))
+
+		// The pipeline version is not allways set, optional.
+		resourceRefs := []*api.ResourceReference{&api.ResourceReference{
+			Key: &api.ResourceKey{
+				Type: api.ResourceType_PIPELINE,
+				Id:   *swf.Spec.PipelineVersionID,
+			},
+		}}
+		if swf.Spec.PipelineVersionID != nil {
+			resourceRefs = append(resourceRefs, &api.ResourceReference{
+				Key: &api.ResourceKey{
+					Type: api.ResourceType_PIPELINE_VERSION,
+					Id:   *swf.Spec.PipelineVersionID,
+				},
+			})
+		}
+
+		_, err := c.runServiceClient.CreateRun(ctx,
+			&api.CreateRunRequest{Run: &api.Run{
+				Name:               workflowName,
+				ResourceReferences: resourceRefs,
+				PipelineSpec: &api.PipelineSpec{
+					PipelineId: *swf.Spec.PipelineID,
+					Parameters: swf.GetParameters(),
+				},
+			}},
+		)
+		if err != nil {
+			return false, "", err
+		}
+
+		return true, workflowName, nil
 
+	}
+
+	newWorkflow, err := swf.NewWorkflow(nextScheduledEpoch, nowEpoch)
```

Comment:
nit: add a comment explaining this is kept for legacy backward compatible support?